### DISCUSSION TITLE
WordPress Multisite Install Documentation

### DISCUSSION THREAD

**Discussion by jbondra** [ddev/ddev#6166] [General]
I recently set up a multisite WordPress install in DDev and found that it worked well enough for subdomain installs provided that `additional_hostnames` in config.yaml was set to:

```
additional_hostnames:
 - "wordpress"
 - "*.wordpress"
```

...though I couldn't get a sub-directory install to work properly unless I added:

```
if (!-e $request_filename) {
        rewrite /wp-admin$ $scheme://$host$request_uri/ permanent;
        rewrite ^(/[^/]+)?(/wp-.*) $2 last;
        rewrite ^(/[^/]+)?(/.*\.php) $2 last;
    }
```

...to nginx_full/nginx-site.conf.

Would it be advisable to add some documentation about WordPress Multisite set-up to the DDev documentation? I can imagine that adding a commented out addition to nginx-site.conf for subdirectory installs would be useful in WordPress projects. Adding a comment in config.yaml for a Wordpress multisite subdomain install might be useful as well.

I managed to figure this out, but I think it'd probably save users a lot of work if this was documented somewhere. I'm not sure if this just isn't a common enough use case but I couldn't seem to find anything regarding WordPress multisite network setups in DDev when searching.

**Comment by rfay** [comment]
We'd love to have docs since you've sorted it out. I'm surprised you have to edit nginx config, but I guess you're essentially using a different docroot? 

Since this is a relatively unusual setup I'd absolutely love to have you do a blog post on ddev.com about it. Alternately, a question-and-answer (click checkbox to answer your own question) on Stack Overflow would be fantastic and have great SEO. We could find a place for it in the ddev docs but it's harder to keep up there. In both of those places you can easily respond to questions and edit it as time goes on.

ddev.com repo: http://github.com/ddev/ddev.com (has full instructions)
Full blog on how to contribute a blog: https://ddev.com/blog/ddev-website-for-contributors/ (has video recording too)

**Reply by jbondra** [reply]
@rfay What the nginx config lacks for a subdirectory install is that slight bit of redirection (in the snippet in my original post) that's rewriting the requests to account for the subdirectory that's tacked on to the URI (e. g. wordpress.ddev.site, wordpress.ddev.site/site1, wordpress.ddev.site/site2, ...etc.). Otherwise, it doesn't redirect properly. I suppose it might help to have that documented somewhere. 

I was slightly incorrect for the subdomain install, in that setup just needs to have `*.wordpress` added to the `additional_hostnames`. The wildcard is needed to account for subdomains (e. g. wordpress.ddev.site, site1.wordpress.ddev.site, site2.wordpress.ddev.site, ...etc). The `wordpress` entry is redundant in my previous post.

Also, since wp-config.php needs some editing to account for the multisite setup you'd also need to remove the `#ddev-generated` comment from wp-config.php to make sure the changes don't get overwritten.

Stupidly, for whatever reason, I didn't realize I could just switch ddev to the `apache-fpm` in that WordPress has support in the UI that will just provide what you need to put in `.htaccess` to account for whichever set up (subdirectory or subdomain) that's needed.

I'll probably start with a Stack Overflow post about this, and I'll consider submitting a blog post about it. The set up really isn't that difficult, I just couldn't find any straightforward documentation or tutorials about it anywhere (and I'd like to have something to refer to out there in case I misplace my personal notes on this stuff...).

I do wonder if there's something that can be done in ddev to automate this set up, or at least make it a little easier to figure out.

**Reply by rfay** [reply]
Sometimes when things have specific apache support that's just the easiest path. Thanks for working on documenting it!

**Reply by astratagem** [reply]
@jbondra thank you for posting!  Since you mentioned some mistakes in the initial post, for the sake of clarity, would you mind editing those examples (if possible)?

Since there's already a section in the docs about Drupal multisite, it would still be great to have the same for WordPress.

I'm going to give the Nginx configuration a shot using the example above, while also referring to the WordPress documentation on the subject (which can, unfortunately, sometimes be misleading/simplistic):

<https://developer.wordpress.org/advanced-administration/server/web-server/nginx/#wordpress-multisite>

**Reply by rfay** [reply]
A WP-specific PR with multisite or other things for https://ddev.readthedocs.io/en/stable/users/usage/cms-settings/#cms-specific-help-and-techniques would be awesome.
